src/cargo/core/compiler/fingerprint/dirty_reason.rs

```diff
@@ -235,6 +256,43 @@ impl DirtyReason {
                             format_args!("the file `{}` has changed ({after})", file.display()),
                         )
                     }
+                    StaleItem::ChangedChecksum {
+                        source,
+                        stored_checksum,
+                        new_checksum,
+                    } => {
+                        let file = source.strip_prefix(root).unwrap_or(&source);
+                        s.dirty_because(
+                            unit,
+                            format_args!(
+                                "the file `{}` has changed (checksum didn't match, {} != {})",
+                                file.display(),
+                                stored_checksum,
+                                new_checksum,
```

Comment:
nit: arg capture whenver possible and making sense

```suggestion
                                "the file `{}` has changed (checksum didn't match, {stored_checksum} != {new_checksum})",
                                file.display(),
```

Reply:
Addressed in ba5899ae0f2211184731b2a8a98313881cb85675